lib/alexa_request_verifier/cert_cache.ex

```diff
@@ -0,0 +1,79 @@
+defmodule AlexaRequestVerifier.CertCache do
+  @moduledoc """
+  This module caches successfully verified Amazon certificates.
+  """
+
+  use GenServer
+
+  require Logger
+
+  @server __MODULE__
+  @table :alexa_request_verifier_certs
+
+  # client
+
+  @doc """
+  Return either the last successfully verified Amazon certificate,
+  or nil if this is the first time the cache has been accessed.
+  """
+  @spec get_cert(cert_url :: String.t()) :: String.t() | nil
+  def get_cert(cert_url) do
+    GenServer.call(@server, {:fetch, cert_url})
+  end
+
+  @doc """
+  Return either the last successfully verified root certificate,
+  or nil if this is the first time the cache has been accessed.
+  """
+  @spec get_last_root() :: String.t() | nil
+  def get_last_root do
+    GenServer.call(@server, {:fetch, "root"})
+  end
+
+  @doc """
+  Store a successfully verified certificate URL in the cache for next time.
+  """
+  @spec store(cert_url :: String.t(), cert_chain :: [String.t()]) :: :ok
+  def store(cert_url, cert_chain) do
+    GenServer.call(@server, {:store, cert_url, cert_chain})
+  end
+
+  @doc """
+  Store a successfully verified root certificate in the cache for next time.
+  """
+  @spec store_root(cert :: String.t()) :: :ok
+  def store_root(cert) do
+    GenServer.call(@server, {:store, "root", cert})
```

Comment:
This is a bad idea because you will have a race condition for multiple requests to write the current "root" certificate in the cache.